modules/tensor_mechanics/src/materials/ComputeVolumetricDeformGrad.C

```diff
@@ -0,0 +1,50 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "ComputeVolumetricDeformGrad.h"
+
+template<>
+InputParameters validParams<ComputeVolumetricDeformGrad>()
+{
+  InputParameters params = validParams<Material>();
+  params.addClassDescription("Computes volumetric deformation gradient and adjusts the total deformation gradient");
+  params.addRequiredParam<MaterialPropertyName>("pre_deform_grad_name", "Name of pre-adjusted deformation gradient variable");
+  params.addRequiredParam<MaterialPropertyName>("volumetric_deform_grad_name", "Name of volumetric deformation gradient variable");
+  params.addRequiredParam<MaterialPropertyName>("post_deform_grad_name", "Name of adjusted deformation gradient variable");
+  return params;
+}
+
+ComputeVolumetricDeformGrad::ComputeVolumetricDeformGrad(const InputParameters & parameters) :
+    DerivativeMaterialInterface<Material>(parameters),
+    _pre_deform_grad(getMaterialProperty<RankTwoTensor>(getParam<MaterialPropertyName>("pre_deform_grad_name"))),
```

Comment:
The nested `getParam<MaterialPropertyName>` calls are not necessary. `getMaterialProperty` does take the input parameter name by default (you can specify a material property name, but then you should use `getMaterialPropertyByName`). So this line should be 

```
    _pre_deform_grad(getMaterialProperty<RankTwoTensor>("pre_deform_grad_name")),
```

And the other lines should be changed accordingly.